db/migrate/20181126123456_create_external_system_configurations.rb

```diff
@@ -0,0 +1,13 @@
+class CreateExternalSystemConfigurations < ActiveRecord::Migration[5.2]
+  def change
+    create_table :external_system_configurations do |t|
+      t.references :external_system, null: false
+      t.string :key, null: false
+      t.string :value, null: false
+      t.boolean :can_read, null: false, default: true
```

Comment:
can_read and can_write were added to allow the UI to show a write only field (password field) where required.  However, problems with activeadmin prevented this from working correctly so at the moment, they are not used but they will be.